lib/jekyll/converters/scss.rb

```diff
@@ -129,6 +185,98 @@ def convert(content)
 
       private
 
+      # The Page instance for which this object acts as a converter.
+      attr_reader :sass_page
+
+      def associate_page_failed?
+        !sass_page
+      end
+
+      # The name of the input scss (or sass) file. This information will be used for error
+      # reporting and will written into the source map file as main source.
+      #
+      # Returns the name of the input file or "stdin" if #associate_page failed
+      def filename
+        return "stdin" if associate_page_failed?
+
+        sass_page.name
+      end
+
+      # The value of the `line_comments` option.
+      # When set to `true` causes the line number and filename of the source be emitted into the
+      # compiled CSS-file. Useful for debugging when the source-map is not available.
+      #
+      # Returns the value of the `line_comments`-option chosen by the user or 'false' by default.
+      def line_comments_option
+        jekyll_sass_configuration.fetch("line_comments", false)
+      end
+
+      # The value of the `sourcemap` option chosen by the user.
+      #
+      # This option controls when sourcemaps shall be generated or not.
+      #
+      # Returns the value of the `sourcemap`-option chosen by the user or ':always' by default.
+      def sourcemap_option
+        jekyll_sass_configuration.fetch("sourcemap", :always).to_sym
+      end
+
+      # Determines whether a sourcemap shall be generated or not.
+      #
+      # Returns `true` if a sourcemap shall be generated, `false` otherwise.
+      def sourcemap_required?
+        return false if associate_page_failed?
+
+        return false if sourcemap_option == :never
+
+        return true if sourcemap_option == :always
+
+        return false if (sourcemap_option == :development) && (Jekyll.env != "development")
+
+        true
+      end
```

Comment:
@Harald-LB Please refactor this method. `sourcemap_required?` is called in 3 different methods.
Every call to those 3 methods are going to cause a flow through these 5 branches each time to return the same result.
This needs to be optimized..

Reply:
I'm a proponent of the "make it work, make it right, make it fast" motto. 
Before optimizing, I'd rather make sure it works as expected.
Last time I tried to save a modification to a Sass file in the dev tools, I encountered an error.

Reply:
@ashmaroli 
Yes, I'll try to add a test, but I will first have to understand how the specs are organized, this might take some time.

About the optimization, I fully agree with Frank. By caching the value of `sourcemap_required?`, we'll save some fractions of a millisecond at best, but we'll have all the trouble to administrate the cached value (for instance it must be reset for every invocation of the converter for a new scss file).

@DirtyF 
What are the "dev tools"? How can I reproduce the error you encountered. Can you provide me with the Jekyll project you have used to test.

Reply:
@Harald-LB Okay. Just adding tests will do. I agree with your point regarding the management of the cached object.